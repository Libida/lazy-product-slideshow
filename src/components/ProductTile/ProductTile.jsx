import React from 'react';

import { Image } from '../Image';
import './ProductTile.scss';

const ProductTile = ({ title, image, imageone }) => {
  const imgSrc = image || imageone;

  return (
    <button className="product-tile">
      <span className="product-tile-inner">
        <Image wrapperClassname="product-tile__img-wrapper" imgClassname="product-tile__img" src={imgSrc}
               alternativeText={title}/>
        <h3 className="product-tile__title">{title}</h3>
      </span>
    </button>
  );
};

export default ProductTile;
