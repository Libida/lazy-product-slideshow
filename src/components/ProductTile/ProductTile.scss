@import './src/styles/helpers';

$img-ratio: 9 / 16;
$title-offset: $gutter-half;

.product-tile {
  margin: 0;
  padding: 0;
  border: 0;
  cursor: pointer;
  animation: append-animate $duration-2 linear;
}

.product-tile-inner {
  display: block;
  position: relative;
  padding-top: $img-ratio * 100%;
  overflow: hidden;
  background-color: $white-1;
}

.product-tile__img-wrapper {
  position: absolute;
  top: 0;
  left: 0;
  width: 100%;
}

.product-tile__title {
  position: absolute;
  left: 0;
  bottom: 0;
  margin: 0;
  padding: $title-offset;
  max-width: calc(100% - #{$title-offset * 2});
  color: $white-1;
  text-shadow: $text-shadow-1;
  box-shadow: $box-shadow-2;
  transition: all 0.4s ease-in-out 0s;

  @extend %text-overflow;
  @extend %heading-1;

  .product-tile:hover & {
    box-shadow: $box-shadow-1;
  }
}


