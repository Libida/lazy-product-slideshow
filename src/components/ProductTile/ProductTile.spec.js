import React from 'react';
import Adapter from '@wojtekmaj/enzyme-adapter-react-17';
import { shallow, configure } from 'enzyme';

configure({ adapter: new Adapter() });

import ProductTile from './ProductTile';

describe('ProductTile', () => {
  const onClickMock = jest.fn();
  const initialProps = {
    title: 'product title',
    image: 'image src',
    url: 'url',
    price: 100,
    discountPrice: 50,
    currency: 'GBP',
    brand: 'Product Brand',
    handleClick: onClickMock,
  };
  const wrapper = (props) => shallow(<ProductTile {...props} />);

  test('should render correctly for default initial props', () => {
    expect(wrapper(initialProps)).toMatchSnapshot();
  });

  test('should render correctly with children', () => {
    expect(wrapper({ ...initialProps, children: <div>Children component</div> })).toMatchSnapshot();
  });

  test('should handle clicks', () => {
    const component = wrapper(initialProps);

    component.simulate('click', onClickMock);
    expect(onClickMock).toHaveBeenCalled();
  });
});
