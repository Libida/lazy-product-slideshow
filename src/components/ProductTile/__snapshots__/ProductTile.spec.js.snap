// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProductTile should render correctly for default initial props 1`] = `
<button
  className="product-tile"
  onClick={[MockFunction]}
>
  <span
    className="product-tile-inner"
  >
    <ProductImage
      alternativeText="product title"
      image="image src"
      imgClassname="product-tile__img"
      wrapperClassname="product-tile__img-wrapper"
    />
    <h3
      className="product-tile__title"
    >
      product title
    </h3>
  </span>
</button>
`;

exports[`ProductTile should render correctly with children 1`] = `
<button
  className="product-tile"
  onClick={[MockFunction]}
>
  <span
    className="product-tile-inner"
  >
    <ProductImage
      alternativeText="product title"
      image="image src"
      imgClassname="product-tile__img"
      wrapperClassname="product-tile__img-wrapper"
    />
    <h3
      className="product-tile__title"
    >
      product title
    </h3>
  </span>
  <div>
    Children component
  </div>
</button>
`;
